File: pointer/#344_reverseString/reverseString.go
package main

import "fmt"

func main() {
	reverseString([]byte{'h','e','l','l','o'})
	reverseString([]byte{'H','a','n','n','a','h'})
	reverseString([]byte{'L','e','e','t','c','o', 'd', 'e'})
}

func reverseString(s []byte)  {
	l := 0
	r := len(s) - 1
	
	for l < r {
		tmp := s[l]
		s[l] = s[r]
		s[r] = tmp
		l++
		r--
	}

	fmt.Println(string(s))
}

File: pointer/#344_reverseString/reverseString.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function (s) {
  let i = 0,
    j = s.length - 1;

  while (i < j) {
    // [s[i], s[j]] = [s[j], s[i]];
    let tmp = s[i];
    s[i] = s[j];
    s[j] = tmp;

    i++;
    j--;
  }

  console.log("ans: ", s);
};

reverseString(["h", "e", "l", "l", "o"]);
reverseString(["H", "a", "n", "n", "a", "h"]);
